hcl2template/types.packer_config.go

```diff
@@ -122,12 +124,22 @@ func (cfg *PackerConfig) EvalContext(ctx BlockContext, variables map[string]cty.
 
 	iterID, ok := cfg.HCPVars["iterationID"]
 	if ok {
+		log.Printf("[WARN] Deprecation: Contextual Variable `iterationID` has been deprecated packer context. " +
```

Comment:
The log here will be problematic, the context is used for evaluating HCL expressions, and we create it often.
The fact that `iterationID` exists does not mean it is used, that happens in other places.
Note: it may not be trivial to figure out with certainty that it's used, as evaluation is done by the HCL library, outside of Packer's code.